pkg/scheduler/framework/statement.go

```diff
@@ -440,3 +441,70 @@ func (s *Statement) Commit() {
 		}
 	}
 }
+
+func (s *Statement) SaveOperations() *Statement {
+	s.outputOperations("Save operations: ", 4)
+
+	stmtTmp := &Statement{}
+	for _, op := range s.operations {
+		stmtTmp.operations = append(stmtTmp.operations, operation{
+			name:   op.name,
+			task:   op.task.Clone(),
+			reason: op.reason,
+		})
+	}
+	return stmtTmp
+}
+
+func (s *Statement) RecoverOperations(stmt *Statement) error {
+	if stmt == nil {
+		return errors.New("statement is nil")
+	}
+	s.outputOperations("Recover operations: ", 4)
+	for _, op := range stmt.operations {
+		switch op.name {
+		case Evict:
+			err := s.Evict(op.task, op.reason)
+			if err != nil {
+				klog.Errorf("Failed to evict task: %s", err.Error())
+				return err
+			}
+		case Pipeline:
+			err := s.Pipeline(op.task, op.task.NodeName, false)
+			if err != nil {
+				klog.Errorf("Failed to pipeline task: %s", err.Error())
+				return err
+			}
+		case Allocate:
+			node := s.ssn.Nodes[op.task.NodeName]
+			err := s.Allocate(op.task, node)
+			if err != nil {
```

Comment:
I think there is a problem here. It is possible that the previous task allocation was successful and the status in the session was changed, but a subsequent task failed. Only the failed task was rolled back and an error was returned. The returned statement has not been committed outside. At this time, it will be inconsistent with the status in the cluster, which will affect the scheduling of subsequent jobs.

Reply:
It will only take effect in session and no commit so cache will not be affected.